common/persistence/cassandra/cassandraHistoryPersistence.go

```diff
@@ -247,10 +244,9 @@ func (h *cassandraHistoryPersistence) GetWorkflowExecutionHistory(request *p.Get
 	return response, nil
 }
 
-func (h *cassandraHistoryPersistence) deserializeEvents(e *p.SerializedHistoryEventBatch) (*p.HistoryEventBatch, error) {
-	p.SetSerializedHistoryDefaults(e)
-	s, _ := h.serializerFactory.Get(e.EncodingType)
-	return s.Deserialize(e)
+func (h *cassandraHistoryPersistence) deserializeEvents(blob *p.DataBlob) (*p.HistoryEventBatch, error) {
+	s, _ := h.serializerFactory.Get(blob.GetEncoding())
```

Comment:
Why did you remove setting of serialized history defaults?  Can this cause problem with existing data?

Reply:
Replied offline. GetEncoding already return Json by default when headers is empty for backward compatibility.